Sources/ProjectDescription/TargetDependency.swift

```diff
@@ -2,12 +2,63 @@ import Foundation
 
 // MARK: - TargetDependency
 
-public enum TargetDependency: Codable {
+/// Dependency status used by `.sdk` target dependencies
+public enum SDKStatus: String {
+    /// Required dependency
+    case required
+
+    /// Optional dependency (weakly linked)
+    case optional
+}
+
+/// Defines the target dependencies supported by Tuist
+public enum TargetDependency: Codable, Equatable {
+    /// Dependency on another target within the same project
```

Comment:
Great idea adding documentation here. This makes developers' life easier when they are editing their manifest files using Xcode.